overlord/hookstate/hooks.go

```diff
@@ -53,6 +54,17 @@ func SetupPostRefreshHook(st *state.State, snapName string) *state.Task {
 	}
 
 	summary := fmt.Sprintf(i18n.G("Run post-refresh hook of %q snap if present"), hooksup.Snap)
+	return HookTask(st, summary, hooksup, nil)
+}
+
+func SetupPreRefreshHook(st *state.State, snapName string) *state.Task {
```

Comment:
As a general comment it'd be nice if we had a no-op thing here when the snap in question doesn't have a particular hook.

Reply:
That applies to the hook machinery, although IMHO it's useful that we have the task logged and visible even if there is no hook ("Run foo hook of bar snap if present" - that's what we see in the tasks). In any case it's a material for separate discussion if we want that.

Reply:
in general we might now even know if the hook is there (because of implicit hooks) when we setup the tasks in a change